### DISCUSSION TITLE
Terraform Module - odd statement

### DISCUSSION THREAD

**Discussion by MrDecisive** [Azure/ipam#80] [General]
Hi
I'm in the process of researching IPAM solutions for our azure infrastructure. I like the idea of an azure native offering so great work on this. We use terraform and I've found there is a third party provider. In the documentation section though it says the provider users tag functionality which is not available in the open source offering? Is there a paid version of this IPAM offering because I can't find a link to it? I'd not what is the statement referring to. Thanks

**Comment by DCMattyG** [comment]
Hi there @MrDecisive, thank you for reaching out on this topic!

First and foremost, there is no paid version of the Azure IPAM project (this repo). It is 100% open source.

Perhaps if you can share which 3rd party TF provider you are reviewing, I can help provide some additional insight into the messaging you're seeing...

As far as Terraform support, there is a Terraform example in the */examples* folder in the repo which shows how to perform a CIDR reservation using Terraform and pipe that into other TF steps. Additionally, we have a customer who has written a full Terraform provider for this project, and will be committing it to the repo soon.

If you need anything else, don't hesitate to reach out. We're here to help in any way we can.

**Reply by MrDecisive** [reply]
@DCMattyG Thanks for the swift reply this is the module I'm looking at. https://registry.terraform.io/providers/XtratusCloud/azureipam/latest/docs

**Comment by DCMattyG** [comment]
Thanks for sharing that @MrDecisive, it looks as though a customer has modified the code we've provided to add some additional TAG data to reservations. I believe this is the same customer we're working with to bring their changes into the upstream project (very soon), and then the TF provider they created will work for all Azure IPAM customers.

In the meantime, we do have a Terraform example in the */examples* folder that may help to get you started until we incorporate their work. And of course, if you need anything, just let us know. We're here to help you be successful!

**Comment by MrDecisive** [comment]
Thanks for that, I'll check out the examples and wait for the full provider
to be merged.

Cheers

**Comment by jeevanions** [comment]
@DCMattyG We are evaluating this Azure IPAM offering and would love to see the Terraform provider. Has this already been published?

**Reply by DCMattyG** [reply]
Hello again @jeevanions, we are currently working with another Azure IPAM customer (mentioned above) whom is in the process of writing this Terraform Provider. It's nearing completion, and will be incorporated into the repo as soon as it's ready. I'd expect it in the next 30 days or so.
